server/routes/ideas.js

```diff
@@ -1,81 +1,268 @@
+
+			const ideaId = isValidObjectId(request.params.ideaId);
+			await ideasData.getIdeaById(ideaId);
+
+			const status = await ideasData.removeIdea(ideaId, user);
+
+			response
+				.status(successStatusCodes.DELETED)
+				.json({
+					status,
+				});
+
+		} catch (e) {
+			sendErrResp(response, e);
+		}
+	});
+
+
+
+// Like / Dislike an Idea
+// ======================================================================
 router
 	.route('/:ideaId/likes')
-	.post(authenticateToken, async (req, res) => {
-		const { user } = req;
+	.post(authenticateToken, async (request, response) => {
+		const { user } = request;
 		try {
 			user._id = isValidObjectId(user._id);
 			user.username = isValidUsername(user.username);
-			const ideaId = isValidObjectId(req.params.ideaId);
+
+			const ideaId = isValidObjectId(request.params.ideaId);
 			const getIdea = await ideasData.getIdeaById(ideaId);
 			if (!getIdea) throw badRequestErr('Could not find any idea with the id');
 			const likeIdeaInfo = await ideasData.likeIdea(ideaId, user);
-			res.status(successStatusCodes.CREATED).json({
-				likeIdeaInfo,
-			});
+
+			response
+				.status(successStatusCodes.CREATED)
+				.json({
+					likeIdeaInfo,
+				});
+
 		} catch (e) {
-			sendErrResp(res, e);
+			sendErrResp(response, e);
 		}
 	})
-	.delete(authenticateToken, async (req, res) => {
-		const { user } = req;
+	.delete(authenticateToken, async (request, response) => {
+		const { user } = request;
 		try {
 			user._id = isValidObjectId(user._id);
 			user.username = isValidUsername(user.username);
-			const ideaId = isValidObjectId(req.params.ideaId);
+
+			const ideaId = isValidObjectId(request.params.ideaId);
 			const getIdea = await ideasData.getIdeaById(ideaId);
 			if (!getIdea) throw badRequestErr('Could not find any idea with the id');
 			const unlikeIdeaInfo = await ideasData.unlikeIdea(ideaId, user);
-			res.status(successStatusCodes.DELETED).json({
-				unlikeIdeaInfo,
-			});
+
+			response
+				.status(successStatusCodes.DELETED)
+				.json({
+					unlikeIdeaInfo,
+				});
+
 		} catch (e) {
-			sendErrResp(res, e);
+			sendErrResp(response, e);
 		}
 	});
 
+
+
+// Create a new Comment for a given Idea
+// ======================================================================
+router
+	.route('/:ideaId/comments')
+	.post(authenticateToken, async (request, response) => {
+		const { user } = request;
+		let { comment } = request.body;
+
+		try {
+			user._id = isValidObjectId(user._id);
+			user.username = isValidUsername(user.username);
+
+			const ideaId = isValidObjectId(request.params.ideaId);
+			await ideasData.getIdeaById(ideaId);
+
+			comment = isValidStr(request.body.comment, 'Comment');
+			const commentObject = {
+				comment,
+				ideaId
+			};
+
+			const comments = await ideasData.createComment(commentObject, user);
+
+			response
+				.status(successStatusCodes.CREATED)
+				.json({
+					comments
+				});
+
+		} catch (e) {
+			sendErrResp(response, e);
+		}
```

Comment:
In createIdea you just added comments for each and every single validations and for createComment you did not.. There is no consistency if possible try to remove comments in every file everybody knows what is happening in the code and what they did.